sys/fido2/ctap/ctap_utils.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup fido2_ctap_utils
+ * @{
+ * @file
+ *
+ * @author     Nils Ollrogge <nils.ollrogge@fu-berlin.de>
+ * @}
+ */
+
+#include <string.h>
+
+#include "xtimer.h"
+
+#include "fido2/ctap.h"
+#include "fido2/ctap/ctap_utils.h"
+
+#if !IS_ACTIVE(CONFIG_FIDO2_CTAP_DISABLE_UP)
+#include "periph/gpio.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if !IS_ACTIVE(CONFIG_FIDO2_CTAP_DISABLE_UP)
+/**
+ * @brief Flag holding information if user is present or not
+ */
+static bool _user_present = false;
+
+/**
+ * @brief Button callback function
+ */
+static void _gpio_cb(void *arg);
+
+int fido2_ctap_utils_init_gpio_pin(void)
+{
+    if (gpio_init_int(BTN0_PIN, BTN0_MODE, GPIO_FALLING, _gpio_cb, NULL) < 0) {
+        return CTAP1_ERR_OTHER;
+    }
+
+    return CTAP2_OK;
+}
+
+int fido2_ctap_utils_user_presence_test(void)
+{
+#ifdef BTN0_PIN
+    int ret;
+
+    gpio_irq_enable(BTN0_PIN);
+
+    if (gpio_init_int(BTN0_PIN, BTN0_MODE, GPIO_FALLING, _gpio_cb, NULL) < 0) {
```

Comment:
This was supposed to move (see [this comment](https://github.com/RIOT-OS/RIOT/pull/16489/files#r674565434))